src/sentry/integrations/slack/event_endpoint.py

```diff
@@ -164,34 +205,60 @@ def on_message(self, request, integration, token, data):
 
         return self.respond()
 
+    def _parse_url(self, link) -> Union[Tuple[LinkType, Mapping[str, Any]], Tuple[None, None]]:
+        """
+        Determines if the URL is something we are able to unfurl, if it is we
+        will return the named groups from the regex match, along with the link
+        type.
+        """
+        link_type, match = match_link(link)
+
+        if link_type is None or match is None:
+            return None, None
+
+        return link_type, match.groupdict()
+
     def on_link_shared(self, request, integration, token, data):
-        parsed_issues = defaultdict(dict)
-        event_id_by_url = {}
+        matches: DefaultDict[LinkType, List[UnfurlableUrl]] = defaultdict(list)
+        links_seen = set()
+
+        # An unfurl may have multiple links to unfurl
         for item in data["links"]:
             try:
-                # we're logging to Kibana because the Slackbot is filtered in Relay
+                # We would like to track what types of links users are sharing,
+                # but it's a little difficult to do in sentry since we filter
+                # requests from Slack bots. Instead we just log to Kibana
                 logger.info(
                     "slack.link-shared", extra={"slack_shared_link": parse_link(item["url"])}
                 )
             except Exception as e:
                 logger.error("slack.parse-link-error", extra={"error": str(e)})
-            event_type, instance_id, event_id = self._parse_url(item["url"])
-            if not instance_id:
+
+            link_type, args = self._parse_url(item["url"])
+
+            if not link_type or args is None:
+                continue
+
+            # Coerce link argument types
+            arg_types = _link_handlers[link_type].arg_types
+            clean_args = {k: arg_types[k](v) if v is not None else None for k, v in args.items()}
```

Comment:
Would it make sense to do this inside `_parse_url`? Feels a little weird to me that we're looking up `arg_types` here.

Reply:
I have an upcoming PR that's going to make this functionality more generic, since there are cases where I need arguments to come out of query parameters too